libmamba/src/core/package_fetcher.cpp

```diff
@@ -433,16 +433,20 @@ namespace mamba
         std::ifstream index_file = open_ifstream(index_path);
         index_file >> index;
 
-        const nlohmann::json solvable_json = m_package_info;
-        index.insert(solvable_json.cbegin(), solvable_json.cend());
+        nlohmann::json repodata_record = m_package_info;
 
-        if (index.find("size") == index.end() || index["size"] == 0)
+        // To take correction of packages metadata (e.g. made using repodata patches) into account,
+        // we insert the index into the repodata record to only add new fields from the index
+        // while keeping the existing fields from the repodata record.
```

Comment:
Great, maybe add the link to the issue it fixes.

Reply:
Done with 0b60e6b948a2d7c1065321166c7ff82dbacab65b.